src/Microsoft.Extensions.Configuration/ConfigurationBuilderExtensions.cs

```diff
@@ -7,9 +7,42 @@
 
 namespace Microsoft.Extensions.Configuration
 {
-    public static class MemoryConfigurationExtensions
+    public static class ConfigurationBuilderExtensions
     {
         /// <summary>
+        /// Includes an existing IConfiguration as a configuration provider to <paramref name="configuraton"/>.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="config">The <see cref="IConfiguration"/> to include.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddConfiguration(this IConfigurationBuilder configurationBuilder, IConfiguration configuration)
+        {
+            if (configurationBuilder == null)
+            {
+                throw new ArgumentNullException(nameof(configurationBuilder));
+            }
+            configurationBuilder.Add(new ChainedConfigurationProvider(configuration));
+            return configurationBuilder;
+        }
+
+        /// <summary>
+        /// Includes an existing IConfiguration as a configuration provider to <paramref name="configuraton"/>.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="config">The <see cref="IConfiguration"/> to include.</param>
+        /// <param name="key">The key of the root of the configuartion to include.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddConfiguration(this IConfigurationBuilder configurationBuilder, IConfiguration configuration, string key)
```

Comment:
Yeah I think that works (assume importing of a section rather than relying
on a prefix).

On Tue, Dec 15, 2015 at 3:29 PM, Hao Kung <notifications@github.com> wrote:

> In
> src/Microsoft.Extensions.Configuration/ConfigurationBuilderExtensions.cs
> <https://github.com/aspnet/Configuration/pull/349#discussion_r47712539>:
>
> > +            if (configurationBuilder == null)
> > +            {
> > +                throw new ArgumentNullException(nameof(configurationBuilder));
> > +            }
> > +            configurationBuilder.Add(new ChainedConfigurationProvider(configuration));
> > +            return configurationBuilder;
> > +        }
> > +
> > +        /// <summary>
> > +        /// Includes an existing IConfiguration as a configuration provider to <paramref name="configuraton"/>.
> > +        /// </summary>
> > +        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
> > +        /// <param name="config">The <see cref="IConfiguration"/> to include.</param>
> > +        /// <param name="key">The key of the root of the configuartion to include.</param>
> > +        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
> > +        public static IConfigurationBuilder AddConfiguration(this IConfigurationBuilder configurationBuilder, IConfiguration configuration, string key)
>
> Cool, so few questions about the prototype. It looks like you only are
> importing top level values, i.e. app.Key = value, and I don't think this
> would pickup child sections like app.Foo:Bar:Key = value. I think we should
> support this if we are doing this as a general feature.
>
> Related to this, if we moved away from app. Prefix, and instead made it a
> specific section, i.e. app:EnableDbTests = true, then this would map to
> importing exactly one config section (perhaps eliminating the need for the
> prefix logic/overload... i.e.
> builder.Include(env.Configuration.GetSection("app"))
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aspnet/Configuration/pull/349/files#r47712539>.
>



-- 
Cheers,
Damian